service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -109,66 +110,20 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		// gcpProject is only used on gcp.
 		gcpProject      = ""
 		gcpProjectFound bool
-		// privateCluster is used only on aws in case the cluster vpc mode is private
-		privateCluster bool
 	)
+	privateCluster, err := privatecluster.IsPrivateCluster(ctx, r.logger, r.k8sClient.CtrlClient(), cr)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
 	{
 		infrastructureRef := cr.Spec.InfrastructureRef
 		if infrastructureRef != nil {
 			switch infrastructureRef.Kind {
 			case infra.AzureClusterKind, infra.AzureManagedClusterKind:
 				provider = infra.AzureClusterKindProvider
-
-				capzCluster := &unstructured.Unstructured{}
-				capzCluster.SetGroupVersionKind(schema.GroupVersionKind{
-					Group:   infrastructureRef.GroupVersionKind().Group,
-					Kind:    infrastructureRef.Kind,
-					Version: infrastructureRef.GroupVersionKind().Version,
-				})
-				err = r.k8sClient.CtrlClient().Get(ctx, client.ObjectKey{
-					Namespace: cr.Namespace,
-					Name:      infrastructureRef.Name,
-				}, capzCluster)
-				if err != nil {
-					return nil, microerror.Mask(err)
-				}
-
-				// TODO: We need to enable this for CAPZ clusters but we first need to understand the implication of this change to the Cilium CNI and the cluster as a whole
-				// blocks := azureCluster.Spec.NetworkSpec.Vnet.CIDRBlocks
-				// if len(blocks) > 0 {
-				//	clusterCIDR = blocks[0]
-				// }
-
-				apiServerLbType, apiServerLbFound, err := unstructured.NestedString(capzCluster.Object, []string{"spec", "networkSpec", "apiServerLB", "type"}...)
-				if err != nil || !apiServerLbFound {
-					return nil, microerror.Mask(fieldNotFoundOnInfrastructureTypeError)
-				}
-
-				privateCluster = apiServerLbType == "Internal"
-
 			case infra.AWSClusterKind, infra.AWSManagedClusterKind:
 				provider = infra.AWSClusterKindProvider
-
-				awsCluster := &unstructured.Unstructured{}
-				awsCluster.SetGroupVersionKind(schema.GroupVersionKind{
-					Group:   infrastructureRef.GroupVersionKind().Group,
-					Kind:    infrastructureRef.Kind,
-					Version: infrastructureRef.GroupVersionKind().Version,
-				})
-				err = r.k8sClient.CtrlClient().Get(ctx, client.ObjectKey{
-					Namespace: cr.Namespace,
-					Name:      infrastructureRef.Name,
-				}, awsCluster)
-				if err != nil {
-					return nil, microerror.Mask(err)
-				}
-
-				annotationValue, annotationFound, err := unstructured.NestedString(awsCluster.Object, []string{"metadata", "annotations", annotation.AWSVPCMode}...)
-				if err != nil || !annotationFound {
-					return nil, microerror.Mask(fieldNotFoundOnInfrastructureTypeError)
-				}
-
-				privateCluster = annotationValue == annotation.AWSVPCModePrivate
 			case infra.VCDClusterKind:
 				provider = infra.VCDClusterKindProvider
 				privateCluster = !reflect.ValueOf(r.proxy).IsZero()
```

Comment:
@whites11 Could you please move this part to the privatecluster.IsPrivateCluster too? Now, if the MC proxy is set, we set privateCluster=true for CAPV and CAPVCD. This is the same check here https://github.com/giantswarm/cluster-apps-operator/pull/463/files#diff-80fcf62ee4658f43e6db43ae61f7948359d478a2c14b42488beea88a2e237793R89 There won't be any change for them in practice and we will have a common method for all providers. 

As far as I see, we don't have privateCluster concept for CAPG so we can move it too and just return false for it.

Reply:
I didn't move those because the private cluster flag depends only on the app config and not the CR. Moving that logic would mean passing to the helper module the app config which seems wrong. The GCP part makes sense though

Reply:
Can you just add some comments that
- In CAPZ and CAPA, we respect WC attributes. A non-private WC can exist in a private MC
- In CAPV and CAPVCD, we respect MC attributes. If MC is private, all WCs are private.
- In CAPG, we don’t have the private concept.

Reply:
discussed on slack